firebase-vscode/src/core/emulators.ts

```diff
@@ -82,17 +82,14 @@ export class EmulatorsController implements Disposable {
     this.emulators.status = "starting";
     this.notifyEmulatorStateChanged();
 
-
+    // TODO: handle rare cases where user cancels on their own, and starts again withing 10 seconds
```

Comment:
Probably just need to save the return value of setTimeou[clearTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout) when we shut down.